www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,442 @@
+import React from "react";
+import { Button, Checkbox, Form, Input, Modal, Space, message, Row, Col, Select, Typography, Divider, Radio, RadioChangeEvent } from "antd";
+import { useStores, Commodity, DraftCreateMineralSite, CandidateEntity, DepositType, Unit, MineralSite } from "models";
+import { LocationInfo } from "../../models/mineralSite/LocationInfo";
+import { Reference, Document } from "../../models/mineralSite/Reference";
+import { GradeTonnage } from "../../models/mineralSite/GradeTonnage";
+import { CommodityStore } from "models/commodity";
+import axios from "axios";
+import { Country, StateOrProvince } from "models";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { UserStore } from "models/user";
+import { MineralInventory, Measure } from "../../models/mineralSite/MineralInventory";
+import { useMemo } from "react";
+
+interface NewMineralSiteModalProps {
+  commodity: Commodity;
+  visible: boolean;
+  onClose: () => void;
+}
+interface FormValues {
+  latitude?: number;
+  longitude?: number;
+  country?: string;
+  stateOrProvince?: string;
+  name: string;
+  refDoc: string;
+  refComment?: string;
+  sourceType: string;
+  depositType: string;
+  depositTypeConfidence: number;
+  tonnage?: number;
+  grade?: number;
+  gradeUnit?: string;
+  tonnageUnit?: string;
+}
+
+export const NewMineralSiteModal: React.FC<NewMineralSiteModalProps> = ({ commodity, visible, onClose }) => {
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+  const commodityOptions = useMemo(() => {
+    return Array.from(commodityStore.records.values()).map((comm) => ({
+      value: comm?.id,
+      label: comm?.name,
+    }));
+  }, [commodityStore.records]);
+  const currentUser = userStore.getCurrentUser()?.url;
+  const countryOptions = useMemo(() => {
+    return Array.from(countryStore.records.values()).map((country) => ({
+      value: country?.id,
+      label: country?.name,
+    }));
+  }, [countryStore.records]);
+
+  const stateOptions = useMemo(() => {
+    return Array.from(stateOrProvinceStore.records.values()).map((state) => ({
+      value: state?.id,
```

Comment:
use URI instead of ID to be consistent with normalizedURI

Reply:
this applies to all predefined entities

Reply:
grade: values.grade
          ? new Measure({
            value: values.grade,
            unit: new CandidateEntity({
              source: currentUser!,
              confidence: values.grade,
              observedName: "%",
              normalizedURI: unitStore.getByURI(values.gradeUnit ? values.gradeUnit : "null")?.id,
            }),
          })
          : undefined,
        ore: values.tonnage
          ? new Measure({
            value: values.tonnage,
            unit: new CandidateEntity({
              source: currentUser!,
              confidence: values.tonnage,
              observedName: "mt",
              normalizedURI: unitStore.getByURI(values.tonnageUnit ? values.tonnageUnit : "null")?.id,
            }),
          })
          : undefined,
        reference: reference,
      });  normalized URI does not have field uri here normalizedURI: unitStore.getByURI(values.tonnageUnit ? values.tonnageUnit : "null")?.id,